<?xml version="1.0"?>
<launch>
	<!-- mode -->
	<!--<arg name="is_tf" default="false" />-->
	<!-- <arg name="is_rviz" default="false" /> -->

	<!-- topic name -->
	<arg name="ndt_pose_topic_name" default="/ndt_pose" />
	<arg name="imu_topic_name" default="/imu/data" />
	<arg name="ekf_pose_topic_name" default="/ekf_pose" />
    <arg name="odom_topic_name" default="/whill/odom" />
	<arg name="velodyne_topic_name" default="/velodyne_points" />
	<arg name="map_topic_name" default="/global_map" />
	<arg name="ndt_pc_topic_name" default="/ndt_pc" />

	<arg name="respawn_pose_topic_name" default="/position/respawn" />

	<!-- frame_id -->
	<arg name="map_frame_id" default="map" />
	<arg name="odom_frame_id" default="odom" />
	<arg name="base_link_frame_id" default="base_link" />
	<arg name="camera_frame_id" default="camera_link" />
	<arg name="velodyne_frame_id" default="velodyne" />

	<!-- ekf -->
	<node pkg="rwrc21_recogni_localizer" type="ekf_node" name="ekf_node" >
		<param name="ndt_pose_topic_name" value="$(arg ndt_pose_topic_name)" />
		<param name="imu_topic_name" value="$(arg imu_topic_name)" />
		<param name="ekf_pose_topic_name" value="$(arg ekf_pose_topic_name)" />
		<param name="odom_topic_name" value="$(arg odom_topic_name)" />
		<param name="map_frame_id" value="$(arg map_frame_id)" />
		<param name="odom_frame_id" value="$(arg odom_frame_id)" />
		<param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
		<param name="is_3DoF" value="true" />
		<param name="is_odom_tf" value="false" />

		<param name="respawn_pose_topic_name" value="$(arg respawn_pose_topic_name)" />

		<rosparam command="load" file="$(find rwrc21_recogni_localizer)/config/ekf/rwrc22_tsukubag.yaml" />
	</node>

	<!-- map_matcher -->
	<node pkg="rwrc21_recogni_localizer" type="map_matcher_node" name="map_matcher_node" output="screen" >
        <param name="pcd_file_path" value="/home/amsl/catkin_ws/src/rwrc22/map/pcd/map_tc.pcd" />
        <!-- <param name="pcd_file_path" value="/home/amsl/pcd/tsukuba/map_tc.pcd" /> -->
		<param name="pc_topic_name" value="$(arg velodyne_topic_name)" />
		<param name="orb_pc_topic_name" value="" />
		<param name="ekf_pose_topic_name" value="$(arg ekf_pose_topic_name)" />
		<param name="ndt_pose_topic_name" value="$(arg ndt_pose_topic_name)" />
		<param name="map_topic_name" value="$(arg map_topic_name)" />
		<param name="ndt_pc_topic_name" value="$(arg ndt_pc_topic_name)" />
		<param name="map_frame_id" value="$(arg map_frame_id)" />
		<param name="is_publish_map" value="true" />
		<param name="is_pcl_offset" value="false" />
		<rosparam command="load" file="$(find rwrc21_recogni_localizer)/config/map_matcher/rwrc22_tsukubag.yaml" />
	</node>
    <include file="$(find velodyne_pointcloud)/launch/VLP-32C_points.launch"/>
    <node pkg="pcl_ros" type="pcd_to_pointcloud" name="pcd_to_pointcloud" args="/home/amsl/catkin_ws/src/rwrc22/map/pcd/map_tc.pcd" output="log">
        <param name="interval" value="0" />
        <remap from="cloud_pcd" to="mapcloud" />
    </node>

	<!-- tf -->
	<!--<group if="$(arg is_tf)" >
		<node pkg="rwrc21_recogni_localizer" type="tf_broadcaster_node" name="tf_broadcaster_node" >
			<param name="is_odom_tf" value="true" />
			<param name="odom_topic_name" value="$(arg odom_topic_name)" />
			<param name="odom_frame_id" value="$(arg odom_frame_id)" />
			<param name="base_link_frame_id" value="$(arg base_link_frame_id)" />
			<param name="camera_frame_id" value="$(arg camera_frame_id)" />
			<param name="velodyne_frame_id" value="$(arg velodyne_frame_id)" />
			<rosparam command="load" file="$(find scan_matching)/config/tf/ikuta.yaml" />
		</node>
	</group>-->

	<!-- rviz -->
    <!-- <group if="$(arg is_rviz)" > -->
	<!-- 	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find rwrc21_recogni_localizer)/config/rviz/rwrc21_tsukuba.rviz" /> -->
	<!-- </group>&#45;&#45;&#62; -->

</launch>
